### PR TITLE
test(e2e): implement QR code helpers

### PR DESCRIPTION
**Pull request by DenysKarmazynDFINITY**
# Motivation

Created a functions for scanning QR code in order to re-use later in e2e tests.

### CODE REVIEW

File: e2e/utils/pages/homepage.page.ts [hunk 1]
```diff
@@ -70,17 +71,6 @@ abstract class Homepage {
...
-		modalOpenButtonTestId,
-		modalTestId
-	}: WaitForModalParams): Promise<Locator> {
-		await this.#page.getByTestId(modalOpenButtonTestId).click();
-		const modal = this.#page.getByTestId(modalTestId);
-		await modal.waitFor();
-
-		return modal;
-	}
-
```
Comment (AntonioVentilii-DFINITY) [inline e2e/utils/pages/homepage.page.ts:83]: is this part related to this PR?

Comment (DenysKarmazynDFINITY) [reply e2e/utils/pages/homepage.page.ts]: Actually no, this change is only needed in my next PR. Removed.

File: e2e/utils/qr-code.utils.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+		width: image.bitmap.width,
+		height: image.bitmap.height
+	};
+
+	const decodedQR = jsQR(imageData.data, imageData.width, imageData.height);
+
+	if (nonNullish(decodedQR)) {
+		return decodedQR.data;
+	}
+};
```
Comment (AntonioVentilii-DFINITY) [inline e2e/utils/qr-code.utils.ts:24]: curiosity for me: does the interpreter say is ok to not return any thing (i.e. undefined)? doesn't he ask to force `return undefined`?

File: e2e/utils/pages/homepage.page.ts [hunk 3]
```diff
@@ -93,6 +94,33 @@ abstract class Homepage {
...
+	private async readQRCode({ selector }: SelectorOperationParams): Promise<string | undefined> {
+		await this.#page.locator(selector).waitFor();
+
+		const dataUrl = await this.getCanvasAsDataURL({ selector });
+
+		if (nonNullish(dataUrl)) {
+			return getQRCodeValueFromDataURL({ dataUrl });
+		}
+	}
+
```
Comment (AntonioVentilii-DFINITY) [inline e2e/utils/pages/homepage.page.ts:123]: same as previous comment

**Review by AntonioVentilii-DFINITY — APPROVED** [review]
LGTM

File: e2e/utils/qr-code.utils.ts (hunk 2 above)
Comment (DenysKarmazynDFINITY) [reply e2e/utils/qr-code.utils.ts:24]: Nope, all good - neither linter nor ts complains here.
